plugin/src/main/kotlin/com/nishtahir/RustAndroidPlugin.kt

```diff
@@ -20,67 +21,104 @@ val toolchains = listOf(
         Toolchain("linux-x86-64",
                 ToolchainType.DESKTOP,
                 "x86_64-unknown-linux-gnu",
-                "<cc>",
-                "<ar>",
+                "<compilerTriple>",
+                "<binutilsTriple>",
                 "desktop/linux-x86-64"),
         Toolchain("darwin",
                 ToolchainType.DESKTOP,
                 "x86_64-apple-darwin",
-                "<cc>",
-                "<ar>",
+                "<compilerTriple>",
+                "<binutilsTriple>",
                 "desktop/darwin"),
         Toolchain("win32-x86-64-msvc",
                 ToolchainType.DESKTOP,
                 "x86_64-pc-windows-msvc",
-                "<cc>",
-                "<ar>",
+                "<compilerTriple>",
+                "<binutilsTriple>",
                 "desktop/win32-x86-64"),
         Toolchain("win32-x86-64-gnu",
                 ToolchainType.DESKTOP,
                 "x86_64-pc-windows-gnu",
-                "<cc>",
-                "<ar>",
+                "<compilerTriple>",
+                "<binutilsTriple>",
                 "desktop/win32-x86-64"),
         Toolchain("arm",
-                ToolchainType.ANDROID,
+                ToolchainType.ANDROID_GENERATED,
                 "armv7-linux-androideabi",
-                "bin/arm-linux-androideabi-clang",
-                "bin/arm-linux-androideabi-ar",
+                "arm-linux-androideabi",
+                "arm-linux-androideabi",
                 "android/armeabi-v7a"),
         Toolchain("arm64",
-                ToolchainType.ANDROID,
+                ToolchainType.ANDROID_GENERATED,
+                "aarch64-linux-android",
+                "aarch64-linux-android",
                 "aarch64-linux-android",
-                "bin/aarch64-linux-android-clang",
-                "bin/aarch64-linux-android-ar",
                 "android/arm64-v8a"),
         Toolchain("x86",
-                ToolchainType.ANDROID,
+                ToolchainType.ANDROID_GENERATED,
+                "i686-linux-android",
+                "i686-linux-android",
                 "i686-linux-android",
-                "bin/i686-linux-android-clang",
-                "bin/i686-linux-android-ar",
                 "android/x86"),
         Toolchain("x86_64",
-                ToolchainType.ANDROID,
+                ToolchainType.ANDROID_GENERATED,
+                "x86_64-linux-android",
+                "x86_64-linux-android",
+                "x86_64-linux-android",
+                "android/x86_64"),
+        Toolchain("arm",
+                ToolchainType.ANDROID_PREBUILT,
+                "armv7-linux-androideabi",
```

Comment:
I know this isn't your work, but a comment mentioning that the differences here are intentional would probably not go amiss.